packages/wasm-dpp/test/unit/document/stateTransition/DocumetsBatchTransition/applyDocumentsBatchTransitionFactory.spec.js

```diff
@@ -16,144 +16,289 @@ const getDocumentTransitionsFixture = require(
-      documentsFixture[0],
-      documents[0],
-      executionContext,
-      executionContext,
+      documentsFixtureJs[0],
+      documentsJs[0],
+      executionContextJs,
+      executionContextJs,
     ]);
 
-    expect(stateRepositoryMock.removeDocument).to.have.been.calledOnceWithExactly(
-      documentTransitions[2].getDataContract(),
-      documentTransitions[2].getType(),
-      documentTransitions[2].getId(),
-      executionContext,
+    expect(stateRepositoryMockJs.removeDocument).to.have.been.calledOnceWithExactly(
+      documentTransitionsJs[2].getDataContract(),
+      documentTransitionsJs[2].getType(),
+      documentTransitionsJs[2].getId(),
+      executionContextJs,
     );
   });
 
+  it('should call `store`, `replace` and `remove` functions for specific type of transitions - Rust', async () => {
+    await applyDocumentsBatchTransition(stateRepositoryMock, stateTransition);
+    expect(stateRepositoryMock.createDocument).to.have.been.calledOnce();
+
+    const [fetchContractId, fetchDocumentType] = stateRepositoryMock.fetchDocuments.getCall(0).args;
+    expect(fetchContractId.toBuffer()).to.deep.equal(documentTransitionsJs[1].getDataContractId());
+    expect(fetchDocumentType).to.equal(documentTransitionsJs[1].getType());
+
+    expect(stateRepositoryMock.updateDocument).to.have.been.calledOnce();
+    expect(stateRepositoryMock.fetchDocuments).to.have.been.calledOnce();
+
+    const [createDocument] = stateRepositoryMock.createDocument.getCall(0).args;
+    const [updateDocument] = stateRepositoryMock.updateDocument.getCall(0).args;
+    const [
+      deleteDataContract,
+      deleteDocumentType,
+      deleteDocumentId,
+    ] = stateRepositoryMock.removeDocument.getCall(0).args;
+
+    expect(createDocument.toObject()).to.deep.equal(documentsFixtureJs[0].toObject());
+    const expectDocument = documentsJs[0].toJSON();
+
+    // ! Why we need to replace. Apparently, `applyDocumentsTransition` somehow
+    // ! modifies `documentsJs` and increments revision and $updatedAt. I have
+    // ! no clue how it's possible.
+    expectDocument.$updatedAt = documentTransitionsJs[1].toJSON().$updatedAt;
+    expectDocument.$revision = documentTransitionsJs[1].toJSON().$revision;
+    expect(updateDocument.toJSON()).to.deep.equal(expectDocument);
+
+    expect(deleteDataContract.toObject()).to.deep.equal(dataContract.toObject());
+    expect(deleteDocumentType).to.deep.equal(documentTransitionsJs[2].getType());
+    expect(deleteDocumentId.toBuffer()).to.deep.equal(documentTransitionsJs[2].getId());
+  });
+
   it('should throw an error if document was not provided for a replacement', async () => {
     fetchDocumentsMock.resolves([]);
 
-    const replaceDocumentTransition = documentTransitions[1];
+    const replaceDocumentTransition = documentTransitionsJs[1];
 
     try {
-      await applyDocumentsBatchTransition(stateTransition);
+      await applyDocumentsBatchTransitionJs(stateTransitionJs);
```

Comment:
We don't need to test old JS functions though, we need to check that the new one produces the same result as the old one

Reply:
fixed - sometimes I leave the original test as a reference - when the rust verison is significanlty modified. But this situation wasn't the case.